stor/src/main/scala/com/avast/clients/storage/stor/StorBackend.scala

```diff
@@ -0,0 +1,162 @@
+package com.avast.clients.storage.stor
+
+import java.io.ByteArrayInputStream
+
+import better.files.File
+import cats.effect.Effect
+import cats.syntax.all._
+import com.avast.clients.storage._
+import com.avast.scala.hashes.Sha256
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.StrictLogging
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.client.blaze.{BlazeClientConfig, Http1Client}
+import org.http4s.headers.`Content-Length`
+import pureconfig.modules.http4s.uriReader
+import pureconfig.{CamelCase, ConfigFieldMapping, ProductHint}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
+import scala.language.higherKinds
+import scala.util.control.NonFatal
+
+trait StorBackend[F[_]] extends StorageBackend[F]
+
+class DefaultStorBackend[F[_]](rootUri: Uri, httpClient: Client[F])(implicit F: Effect[F]) extends StorBackend[F] with StrictLogging {
+
+  override def head(sha256: Sha256): F[Either[StorageException, HeadResult]] = {
+    logger.debug(s"Checking presence of file $sha256 in Stor")
+
+    try {
+      val request = Request[F](
+        Method.HEAD,
+        rootUri / sha256.toString
+      )
+
+      httpClient.fetch(request) { resp =>
+        resp.status match {
+          case Status.Ok =>
+            `Content-Length`.from(resp.headers) match {
+              case Some(`Content-Length`(length)) => F.pure(Right(HeadResult.Exists(length)))
+              case None =>
+                resp.bodyAsText.compile.last.map { body =>
+                  Left(StorageException.InvalidResponseException(resp.status.code, body.toString, "Missing Content-Length header"))
+                }
+            }
+          case Status.NotFound =>
+            F.pure(Right(HeadResult.NotFound))
+
+          case _ =>
+            resp.bodyAsText.compile.last.map { body =>
+              Left(StorageException.InvalidResponseException(resp.status.code, body.toString, "Unexpected status"))
+            }
+        }
+      }
+    } catch {
+      case NonFatal(e) => F.raiseError(e)
```

Comment:
Why not `F.pure(Left(...))`

Reply:
Well, this is kind of a question. Should be throwing an _unknown_ exception considered as fatal failure or as normal state? ;-)